overlord/snapstate/handlers.go

```diff
@@ -3743,6 +3743,28 @@ func (m *SnapManager) undoMigrateSnapHome(t *state.Task, tomb *tomb.Tomb) error
 	return writeMigrationStatus(t, snapst, snapsup)
 }
 
+func (m *SnapManager) doUpdateValidationSetTracking(t *state.Task, _ *tomb.Tomb) error {
+	st := t.State()
+	st.Lock()
+	defer st.Unlock()
+
+	var validationSets []string
+	if err := t.Get("validation-sets", &validationSets); err != nil {
+		return err
+	}
+
+	snaps, ignoreValidation, err := InstalledSnaps(st)
+	if err != nil {
+		return err
+	}
+
+	if err := TryEnforceValidationSets(st, validationSets, 0, snaps, ignoreValidation); err != nil {
```

Comment:
I don't remember the details but the doc of this says it fetches validation sets. Don't we risk to fetch newer ones than the one we fetched originally in daemon? do we carry more information to this point like exact sequence point, if we don't we have that issue?

Reply:
Yes, I've added a new function for local-only enforcement that should fix this. I'll add something to the backlog to refactor this part and change the validation set code to be less repetitive and more reusable